### PR TITLE
python3Packages.alectryon: patch to fix permission issues

### PR DESCRIPTION
**Pull request by Zimmi48**
###### Motivation for this change

Thanks to @cpitclaudel for writing this patch! This avoids the issue I was describing at https://github.com/NixOS/nixpkgs/pull/134454.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @siraben

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134529` at 6f31f848 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.alectryon</li>
    <li>python39Packages.alectryon</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/python-modules/alectryon/default.nix:13:15:
```
   |
13 |   patches = [ ./c779def3fa268e703d4e0ff8ae0b2981e194b269.patch ];
   |               ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/alectryon/default.nix:27:0:
```
   |
27 |     description = "A collection of tools for writing technical documents that mix Coq code and prose";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/alectryon/c779def3fa268e703d4e0ff8ae0b2981e194b269.patch [hunk 1]
```diff
@@ -0,0 +1,37 @@
+From c779def3fa268e703d4e0ff8ae0b2981e194b269 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Cl=C3=A9ment=20Pit-Claudel?= <clement.pitclaudel@live.com>
+Date: Tue, 17 Aug 2021 11:29:16 -0400
+Subject: [PATCH] cli: Don't copy permissions of asset files
+
+* alectryon/cli.py (copy_assets): Use shutil.copyfile (which does not copy file
+permissions) instead of shutil.copy.
+
+https://github.com/NixOS/nixpkgs/pull/134454
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/alectryon/c779def3fa268e703d4e0ff8ae0b2981e194b269.patch:9]: Can we fetch this PR instead?

Comment (Zimmi48) [reply pkgs/development/python-modules/alectryon/c779def3fa268e703d4e0ff8ae0b2981e194b269.patch:9]: @SuperSandro2000: this PR is just the place where I reported the issue to the upstream maintainer.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/alectryon/c779def3fa268e703d4e0ff8ae0b2981e194b269.patch:9]: did you submit it to the alectryon project? Then we could fetch that commit? If not please name the patch better than the commit sha.

Comment (Zimmi48) [reply pkgs/development/python-modules/alectryon/c779def3fa268e703d4e0ff8ae0b2981e194b269.patch:9]: The commit was not authored by myself but by the project maintainer. So yes, we can fetch the commit from the project if that's considered better practice.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/alectryon/c779def3fa268e703d4e0ff8ae0b2981e194b269.patch:9]: Yes, please do that.

Comment (Zimmi48) [reply pkgs/development/python-modules/alectryon/c779def3fa268e703d4e0ff8ae0b2981e194b269.patch]: Done.

File: pkgs/development/python-modules/alectryon/default.nix [hunk 2]
```diff
@@ -10,6 +11,11 @@ buildPythonPackage rec {
     sha256 = "sha256:0mca25jv917myb4n91ccpl5fz058aiqsn8cniflwfw5pp6lqnfg7";
   };
 
+  patches = [ (fetchpatch {
+    url = "github.com/cpitclaudel/alectryon/commit/c779def3fa268e703d4e0ff8ae0b2981e194b269.patch";
+    sha256 = "sha256:0xsz56ibq8xj7gg530pfm1jmxbxw4r6v8xvzj5k1wdry83srqi65";
+  } ) ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/alectryon/default.nix:17]:
```suggestion
  patches = [
    (fetchpatch {
      url = "https://github.com/cpitclaudel/alectryon/commit/c779def3fa268e703d4e0ff8ae0b2981e194b269.patch";
      sha256 = "0xsz56ibq8xj7gg530pfm1jmxbxw4r6v8xvzj5k1wdry83srqi65";
     })
   ];
```

**Pull request merged**